lib/topological_inventory/core/service_catalog_client.rb

```diff
@@ -0,0 +1,53 @@
+require "rest_client"
+require "topological_inventory/core/service_plan_client"
+
+module TopologicalInventory
+  module Core
+    class ServiceCatalogClient
+      def initialize(source)
+        @source = source
+      end
+
+      # TODO? Move this to a module in the API repo
+      def order_service_plan(plan_name, service_offering_name, additional_parameters)
+        payload = ServicePlanClient.new.build_payload(
+          plan_name, service_offering_name, catalog_parameters(additional_parameters)
+        )
+        response = make_request(:post, order_service_plan_url, payload)
+        JSON.parse(response.body)
+      end
+
+      private
+
+      def order_service_plan_url
+        URI.join(@source.default_endpoint.base_url_path, "apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances").to_s
+      end
+
+      def make_request(method, url, payload, headers = generic_headers)
+        request_options = {
+          :method     => method,
+          :url        => url,
+          :headers    => headers,
+          :verify_ssl => @source.default_endpoint.verify_ssl,
```

Comment:
We made verify_ssl a boolean here instead of an int so you'll have to build the `OpenSSL::SSL::VERIFY_*` yourself.
Something like

`:verify_ssl => @source.default_endpoint.verify_ssl ? OpenSSL::SSL::VERIFY_PEER : OpenSSL::SSL::VERIFY_NONE`
We can deal with a custom certificate_authority later